fairlearn/metrics/true_positive_rate.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+
+from . import MetricsResult
+
+
+def true_positive_rate(y_actual, y_predict, group_id):
+    result = MetricsResult()
+
+    groups = np.unique(group_id)
+
+    positives_all = 0
+    true_positives_all = 0
+
+    positives = {}
+    true_positives = {}
+    for group in groups:
+        positives[group] = 0
+        true_positives[group] = 0
+
+    for y, yhat, a in zip(y_actual, y_predict, group_id):
```

Comment:
Leaving aside whether this is the best way to implemented the loop in Python (answer: almost certainly not), there's a wider point.

Calculating metrics like this (single request per metric) is going to result in lots of scans through the data. That's going to be slow; the same loop could gather up true negatives, false positives and false negatives essentially for free.

Reply:
Have had a discussion with @rihorn2 and @imatiach-msft  with a conclusion that I'm worrying prematurely about Big Data in the above. For the data sizes we're going to consider, shifting this to using numpy vector operations will be far more important to performance than worrying about unnecessary scans through the data.